price_parser/parser.py

```diff
@@ -52,8 +53,20 @@ def fromstring(cls, price: Optional[str],
 
 
 def or_regex(symbols: List[str]) -> Pattern:
-    """ Return a regex which matches any of ``symbols`` """
-    return re.compile('|'.join(re.escape(s) for s in symbols))
+    """ Return a regex which matches any of ``symbols`` surrounded by some special characters """
+    left_tokens = [r"^", r"\s+?", r"\d+?"]
+    right_tokens = [r"$", r"\s+?", r"\d+?", r"[^a-zA-Z0-9]+?"]
+
+    return re.compile(
+        "|".join(
+            [
+                left + "({})".format(re.escape(s)) + right
+                for left, right in itertools.product(left_tokens, right_tokens)
+                for s in symbols
+            ]
+        )
+    )
```

Comment:
I tested and it didn't work in all examples.
I don't consider simple any code that auto generate regexes. Your code may work but it will require more effort on it to solve all test cases and we could try to improve it, but for now, I believe that solving the existing issues is more interesting for the project.